src/functions/asyncApi2PayloadValidation.ts

```diff
@@ -0,0 +1,78 @@
+import { ValidateFunction } from 'ajv';
+
+import { IFunction, IFunctionResult } from '../types';
+import { getLintTargets } from '../utils';
+import { schema } from './schema';
+import { ISchemaFunction } from './schema';
+
+import * as asyncApi2Schema from '../rulesets/asyncapi/schemas/schema.asyncapi2.json';
+
+export interface IPayloadValidationOptions {
+  field?: string;
+}
+
+const fakeSchemaObjectId = 'asyncapi2#schemaObject';
+const asyncApi2SchemaObject = { $ref: fakeSchemaObjectId };
+
+let validator: ValidateFunction;
+
+const buildAsyncApi2SchemaObjectValidator = (schemaFn: ISchemaFunction): ValidateFunction => {
+  if (validator !== void 0) {
+    return validator;
+  }
+
+  const ajv = schemaFn.createAJVInstance({
+    meta: false,
+    jsonPointers: true,
+    allErrors: true,
+  });
+
+  ajv.addMetaSchema(schema.specs.v7);
+  ajv.addSchema(asyncApi2Schema, asyncApi2Schema.$id);
+
+  validator = ajv.compile(asyncApi2SchemaObject);
+
+  return validator;
+};
+
+export const asyncApi2PayloadValidation: IFunction<IPayloadValidationOptions> = (
+  targetVal,
+  opts,
+  paths,
+  otherValues,
+) => {
+  const ajvValidationFn = buildAsyncApi2SchemaObjectValidator(schema);
+
+  const relevantItems = getLintTargets(targetVal, opts.field);
```

Comment:
Just for the clarity, we've discussed this one on slack - this should not be needed, since `runner` already handles it for us.